reconciletags/reconciletags_e2e_test.py

```diff
@@ -0,0 +1,93 @@
+"""End to end test for the tag reconciler."""
+
+import json
+import os
+import subprocess
+import sys
+import unittest
+
+bucket = ''
+
+
+class ReconciletagsE2eTest(unittest.TestCase):
+
+    _FILE_NAME = "e2e_test.json"
+    _DIR = 'tiny_docker_image/'
+    _REPO = 'gcr.io/google-appengine-qa/e2etest'
+    _TAG = 'initial'
+    _TEST_JSON = """
+    {{
+      "projects":[
+        {{
+          "registries": ["gcr.io"],
+          "repository": "google-appengine-qa/e2etest",
+          "images": [
+                  {{
+                      "digest": "{0}",
+                      "tag": "testing"
+                  }}
+          ]
+        }}
+      ]
+    }}
+    """
+
+    def _ListTags(self, repo):
+        output = json.loads(
+            subprocess.check_output(['gcloud', 'beta', 'container',
+                                     'images', 'list-tags',
+                                     '--no-show-occurrences',
+                                     '--format=json', repo]))
+        return output
+
+    def _BuildImage(self, full_image_name, bucket):
+        # create a non-functional but tiny docker image
+        subprocess.call(['gcloud', 'beta', 'container', 'builds',
+                         'submit', self._DIR, '-q', '--tag', full_image_name,
+                         '--gcs-log-dir', bucket + '/logs',
+                         '--gcs-source-staging-dir', bucket + '/staging'])
+
+        # create the json config file and write to it
+        test_json = open(self._FILE_NAME, 'w')
+
+        # grab the just created digest
+        output = self._ListTags(self._REPO)
+        digests = [image['digest'].split(':')[1] for image in output]
+        self.assertEqual(len(digests), 1)
+        self.digest = digests[0]
```

Comment:
nit: i think this looks a bit clearer if you check the number of image tags instead of the number of digests.

```
output = self._ListTags(self._REPO)
self.assertEqual(len(output, 1))
self.digest = output.pop().split(':')
```